ui/index.js

```diff
@@ -128,8 +128,10 @@ async function startApp(metamaskState, backgroundConnection, opts) {
     );
   }
 
-  backgroundConnection.on('update', function (state) {
-    store.dispatch(actions.updateMetamaskState(state));
+  backgroundConnection.onNotification((data) => {
+    if (data.method === 'sendUpdate') {
+      store.dispatch(actions.updateMetamaskState(data.params[0]));
+    }
```

Comment:
Nit: might be worth adding an `else` case and throwing an error, in case we screw something up and send something else as a notification.

Reply:
not sure I like how prettier is breaking up this over multiple lines with the backticks :(